src/hb-directwrite.cc

```diff
@@ -854,6 +854,9 @@ hb_directwrite_face_create (IDWriteFontFace *font_face)
 IDWriteFontFace *
 hb_directwrite_face_get_font_face (hb_face_t *face)
 {
+  if (face->destroy == _hb_directwrite_font_release)
+    return (IDWriteFontFace *) face->user_data;
+
   return face->data.directwrite->fontFace;
```

Comment:
Should the final return just be NULL?

Under what circumstances would you fail the if condition, but have a non-NULL face->data.directwrite->fontFace ?

Reply:
If the font was NOT constructed from a directwrite object... In which case we try to create a dwrite object. Maybe we should separate that as a different API? Or add new API for just returning the original object?

Reply:
CoreText has the same issue currently.

Reply:
> Comparing to the coretext code, I don't know - that code doesn't do any checks, and seems to just assume that data.coretext exists ?

data.* is automatically and lazily created...

Reply:
As I said, CoreText is also broken in various ways.